File: jessie/android-hook/src/main/java/android/os/ServiceManagerHook.java
package android.os;

import java.util.Map;

@SuppressWarnings("ALL")
public final class ServiceManagerHook {

    public static IBinder getService(String name) {
        return null;
    }

    public static IBinder getServiceOrThrow(String name) {
        return null;
    }

    public static void addService(String name, IBinder service) {
    }

    public static void addService(String name, IBinder service, boolean allowIsolated) {
    }

    public static IBinder checkService(String name) {
        return null;
    }

    public static String[] listServices() {
        return null;
    }

    public static void initServiceCache(Map<String, IBinder> cache) {
    }
}

File: jessie/src/main/java/android/os/ServiceManagerHook.kt
package android.os

import cn.jessie.etc.Logdog

object ServiceManagerHook {
    @JvmStatic
    fun getService(name: String): IBinder? {
        Logdog.error("ServiceManagerHook.getService($name)")
        return ServiceManager.getService(name)
    }

    @JvmStatic
    fun getServiceOrThrow(name: String): IBinder? {
        return ServiceManager.getServiceOrThrow(name)
    }

    @JvmStatic
    fun addService(name: String, service: IBinder) {
        ServiceManager.addService(name, service)
    }

    @JvmStatic
    fun addService(name: String, service: IBinder, allowIsolated: Boolean) {
        ServiceManager.addService(name, service, allowIsolated)
    }

    @JvmStatic
    fun checkService(name: String): IBinder? {
        return ServiceManager.checkService(name)
    }

    @JvmStatic
    fun listServices(): Array<String>? {
        return ServiceManager.listServices()
    }

    @JvmStatic
    fun initServiceCache(cache: Map<String, IBinder>) {
        ServiceManager.initServiceCache(cache)
    }
}
